### DISCUSSION TITLE
Long running Background task

### DISCUSSION THREAD

**Discussion by fredski02** [supabase/supabase#4661] [General]
I'm thinking about using supabase and part of my idea for a webapp I had in mind requires a backend task that could span a long time.

The crux of the task is that players/users each have 1 minute to choose some tokens and do so in a turn based fashion - once user A has picked, user B is given 1 minute to pick and then user C is given 1 minute to pick. The process repeats itself until all the tokens have been picked.

It occurred to me that this might be hard to model and carry out in a traditional SQL table. Sure, you can say when the token picking should begin and who's turn it is to pick but what if the player leaves or the time runs out? I want to auto assign them a token and move to the next user.

Perhaps it could be modelled in a normal PostGres table, any thoughts or work arounds?

Thanks,

Freddie

**Comment by revskill10** [comment]
Is there any update ?

**Comment by bombillazo** [comment]
Do you have any insights into how to do a long-running process natively in the Supabase stack? Or are we required to use another service/stack for these?
